### PR TITLE
build tests: check that Github pages can be built successfully

### PR DESCRIPTION
**Pull request by bluca**

### CODE REVIEW

File: meson.build [hunk 1]
```diff
@@ -3501,6 +3501,19 @@ meson.add_install_script('sh', '-c', 'touch $DESTDIR@0@'.format(prefixdir))
...
 
+# Ensure that changes to the docs/ directory do not break the
+# basic Github pages build.
+jekyll = find_program('jekyll', required : false)
+if want_tests != 'false' and jekyll.found()
+        test('github-pages',
+             jekyll,
+             args : ['build',
+                     '--source', join_paths(project_source_root, 'docs'),
+                     '--destination', join_paths(project_build_root, '_site')])
```
Comment (yuwata) [inline meson.build:3514]: Just a question, why `'_site'`?

**Review by yuwata — APPROVED** [review]
LGTM.

File: meson.build (hunk 1 above)
Comment (bluca) [reply meson.build:3514]: it's the default name that the `jekyll` tool uses: https://manpages.debian.org/unstable/jekyll/jekyll.1.en.html#OPTIONS

Comment (yuwata) [reply meson.build:3514]: Ah, I see. Thank you.

Comment (keszybz) [reply meson.build:3514]: It uses `_` to not conflict with other user files. But we could use whatever we want. Or more precisely, anything except `sysvinit` and `var-log` which are currently used,

**Comment by keszybz** [comment]
Hmm, I think we should limit this under "developer mode". This test makes no sense for end users, and I'm worried that it could easily break with different jekyll versions and whatnot.

**Comment by bluca** [comment]
> Hmm, I think we should limit this under "developer mode". This test makes no sense for end users, and I'm worried that it could easily break with different jekyll versions and whatnot.

does the unit tests GH action run in developer mode?

**Comment by bluca** [comment]
> > Hmm, I think we should limit this under "developer mode". This test makes no sense for end users, and I'm worried that it could easily break with different jekyll versions and whatnot.
> 
> does the unit tests GH action run in developer mode?

self-answer: it does. Updated, ptal

**Comment by yuwata** [comment]
LGTM.

**Pull request merged**